app/services/digest_scheduler_service.rb

```diff
@@ -0,0 +1,29 @@
+module DigestSchedulerService
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def call(*args)
+      new.call(*args)
+    end
+  end
+
+  def call
+    run_with_advisory_lock do
+      digest_run = DigestRun.find_or_initialize_by(
```

Comment:
I don't know if this is related to this PR, but if someone were to accidentally (or intentionally to deal with an incident?) call the `WeeklyDigestSchedulerService` one day after it had ran, it would end up creating a new `DigestRun` starting one day after rather than finding the existing one suitable for that week. I don't know if this is a concern of this class, or something more higher up?

Reply:
I think we were talking about doing a spike into what we would do in those instances so I've just made it as simple as possible here and not worried about it.